therapi-runtime-javadoc/src/main/java/com/github/therapi/runtimejavadoc/internal/parser/CommentParser.java

```diff
@@ -0,0 +1,84 @@
+package com.github.therapi.runtimejavadoc.internal.parser;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.github.therapi.runtimejavadoc.Comment;
+import com.github.therapi.runtimejavadoc.CommentElement;
+import com.github.therapi.runtimejavadoc.CommentText;
+import com.github.therapi.runtimejavadoc.InlineLink;
+import com.github.therapi.runtimejavadoc.InlineTag;
+import com.github.therapi.runtimejavadoc.Link;
+
+class CommentParser {
+
+    private static final Pattern inlineTag = Pattern.compile("\\{@(\\w+)\\s+(\\w[^}]+)\\}");
+
+    private static final Pattern inlineTagSplitter = Pattern.compile("((?=(\\{@))|(?<=}))");
+
+    private static final Pattern whitespace = Pattern.compile("\\s+");
+
+    private static final Pattern linkRefSplitter = Pattern.compile("#");
+
+    static Comment parse(String commentText) {
+        String[] parts = inlineTagSplitter.split(commentText);
```

Comment:
IMO this could be better solved with a inlineTag matcher find startIndex endIndex search. If you need an example I can write it later today.

Reply:
Indeed this is a better approach, it was around 3-4am, probably why I used lookaheads instead !

Reply:
Done.